src/openklant/management/commands/migrate_to_v2_phonenumbers.py

```diff
@@ -0,0 +1,244 @@
+)
+
+MIGRATION_TOKEN_IDENTIFIER = "Migration application"
+
+
+def _retrieve_klanten(url: str, access_token: str) -> Tuple[list[Klant], str | None]:
+    klanten_path = "/klanten/api/v1/klanten"
+
+    _url = urlparse(url)
+    _params = parse_qs(_url.query)
+
+    client = LegacyOpenKlantClient(f"{_url.scheme}://{_url.netloc}", access_token)
+    response_data = client.retrieve(_url.path or klanten_path, params=_params)
+
+    if not isinstance(response_data, dict):
+        logger.error("unexpected_response_data", response_data=response_data)
+        return [], None
+
+    items = response_data.get("results", [])
+    klanten = []
+
+    generic_client = Client()
+
+    for data in items:
+        klant = Klant(
+            **{field: value for field, value in data.items() if field in KLANT_FIELDS}
+        )
+
+        if klant.subject and not klant.subject_type:
+            subject_data = generic_client.retrieve(klant.subject)
+
+            if not isinstance(subject_data, dict):
+                logger.error(
+                    "unexpected_response_data",
+                    context="subject_retrieval",
+                    subject_data=subject_data,
+                )
+
+                continue
+
+            klant.set_from_external_subject(subject_data)
+
+        klanten.append(klant)
+
+    next_url = response_data.get("next")
+    return klanten, next_url
+
+
+NUMMER_FIELD_MAPPING = {
+    "natuurlijk_persoon": "inp_bsn",
+    "niet_natuurlijk_persoon": "inn_nnp_id",
+    "vestiging": "vestigings_nummer",
+}
+
+
+def _save_klanten_phonenumbers(
+    url: str, token: TokenAuth, klanten: list[Klant]
+) -> list[str]:
+    if not klanten:
+        logger.info("no_klanten_to_save")
+        return []
+
+    logger.info("updating_klanten_v2_api", count=len(klanten))
+
+    openklant_client = OpenKlantClient(url, token)
+    updated_klanten = []
+
+    for klant in klanten:
+        subject_type = getattr(klant, "subject_type", None)
+        nummer_field = NUMMER_FIELD_MAPPING.get(subject_type)
+        nummer = getattr(klant.subject_identificatie, "get", lambda x: None)(
+            nummer_field
+        )
+        try:
+            soort_partij = klant._get_soort_partij()
+            partij = Partij.objects.get(soort_partij=soort_partij, nummer=nummer)
+        except Partij.DoesNotExist:
+            logger.warning(
+                "partij_not_found",
+                klant=klant,
+                subject_type=subject_type,
+                nummer=nummer,
+            )
+            continue
+        existing_da = DigitaalAdres.objects.filter(
+            partij=partij,
+            soort_digitaal_adres="telefoonnummer",
+            referentie="portaalvoorkeur",
+        ).first()
+
+        if existing_da:
+            logger.info(
+                "digitaaladres_exists",
+                partij=partij.pk,
+                nummer=nummer,
+            )
+            continue
```

Comment:
Currently this does up to two queries per `klant` in `klanten`, which can become quite slow if there is a decent number of klanten.

You could simplify this by doing a single query on `Partij` to figure out which `Partijen` already have a telefoonnummer marked as portaalvoorkeur (something like `Partij.objects.filter(digitaaladres__referentie="portaalvoorkeur", digitaaladres__soort_digitaal_adres=SoortDigitaalAdres.telefoonnumer)`) and then you could skip those partijen in the loop